### PR TITLE
Fix the `is_successful` assertion corner case

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
This was breaking my attempts to write tests that did `job.assert_status(['successful', 'canceled'])`. Because "successful" is in the allowed status list, it won't pass on the "canceled" status because `is_successful` isn't true.

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - API

### CODE REVIEW

File: awxkit/awxkit/api/mixins/has_status.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def assert_status(self, status_list, msg=None):
             status_list = [status_list]
         if self.status in status_list:
             # include corner cases in is_successful logic
-            if 'successful' not in status_list or self.is_successful:
+            if self.status != 'successful' or self.is_successful:
```
Comment (jbradberry) [inline awxkit/awxkit/api/mixins/has_status.py:66]: This doesn't seem useful, `is_successful` is literally defined as `return self.status == 'successful'`.  It seems to me that this line then will always be true.

If I follow what is going on, this changes the intent of this block.

**Review by jbradberry — CHANGES_REQUESTED** [review]
To be honest, looking at this I realize that this sub-block should probably be taken out entirely.  If `self.status in status_list` is true, we ought to just bail out immediately.

**Comment by jbradberry** [comment]
@AlanCoding having just tracked down the original PR that put that there, which indicates that `is_successful` may have different implementations, I am amending my opinion.  Something like this seems called for:

```
if ('successful' in status_list) == self.is_successful:
    return
```

This sub-block needs a bit better of a comment, though.  We shouldn't have to dig through a stale repo to find this out.

**Comment by AlanCoding** [comment]
```python
if ('successful' in status_list) == self.is_successful:
    return
```

This still doesn't allow the fairly reasonable use of `job.assert_status(['successful', 'canceled'])`, because a canceled job is _not_ successful. That's why I think this needs to reference `self.status` in some way. The conditional is hard to read, that's true.

If I go through all the `is_successful` definition, I honestly don't find any of them helpful looking, and I wouldn't be opposed to deleting this block.

**Comment by jbradberry** [comment]
Yeah, no, I just realized that my suggestion isn't workable either.  This is tricky, and I don't think anymore that getting rid of the inner block is the way to go either.

**Comment by jbradberry** [comment]
Clearing out my review, in case this gets picked up again.

**Comment by AlanCoding** [comment]
Clearly, this is challenging to think about. Although believed to be valid, I will close because unlikely to have time for it.
